### ISSUE TITLE
ScrollMetrics.pixels have problem

### ISSUE THREAD

**Issue by mofada** [flutter/flutter#50560]
## question
I have a need, my expected effect is like this.

![image](https://user-images.githubusercontent.com/22286246/74241213-84bfc100-4d16-11ea-8c2c-1065c0cd94bd.png)

Then I also achieved this effect, but I have a problem, you can see the picture below, if I slide slowly, it is normal, if I slide very fast, problems will occur. I need your help.

![slow](https://user-images.githubusercontent.com/22286246/74241876-dc126100-4d17-11ea-8273-08e36cfd5fb8.gif)

![image](https://user-images.githubusercontent.com/22286246/74242384-c3567b00-4d18-11ea-8feb-2bd0478f3fe0.png)

This is a quick swipe

![quickly](https://user-images.githubusercontent.com/22286246/74242000-18de5800-4d18-11ea-8a6a-d681f68a6fe0.gif)

![image](https://user-images.githubusercontent.com/22286246/74242638-437ce080-4d19-11ea-9e8e-5c0999113208.png)

## code
I wrote this code after PageView, because PageView does not meet my needs, so I can only rewrite it myself.

### schedule_list_view.dart
```
import 'dart:math';

import 'package:flutter/cupertino.dart';
import 'package:flutter/gestures.dart';
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

/// @author : fada
/// @email : fada@mofada.cn
/// @date : 2019/10/8 11:24
/// @description : input your description

class ScheduleController extends ScrollController {
  ///item 宽度
  final double itemWidth;

  ///每次缩小30dp
  final double differentWidth;

  ///缩小数量
  final int differentCount;

  ///item间距
  final double itemSpace;

  ScheduleController({
    this.itemSpace = 20,
    this.itemWidth = 160,
    this.differentWidth = 30,
    this.differentCount = 4,
  });

  ///务必重写此方法, 否则创建默认[ScrollPositionWithSingleContext]
  @override
  ScrollPosition createScrollPosition(ScrollPhysics physics,
      ScrollContext context, ScrollPosition oldPosition) {
    return _SchedulePosition(
      physics: physics,
      context: context,
      oldPosition: oldPosition,
      itemWidth: itemWidth,
      itemSpace: itemSpace,
      differentCount: differentCount,
      differentWidth: differentWidth,
    );
  }

  @override
  void attach(ScrollPosition position) {
    super.attach(position);
    final _SchedulePosition pagePosition = position;
    //todo ? 为什么要赋值? 看注释貌似是跳转页面会调用这个方法
    pagePosition.itemSpace = itemSpace;
    print(position);
  }
}

class ScheduleMetrics extends FixedScrollMetrics {
  ScheduleMetrics({
    @required double minScrollExtent,
    @required double maxScrollExtent,
    @required double pixels,
    @required double viewportDimension,
    @required AxisDirection axisDirection,
    @required this.itemWidth,
    @required this.differentWidth,
    @required this.differentCount,
    @required this.itemSpace,
  }) : super(
          minScrollExtent: minScrollExtent,
          maxScrollExtent: maxScrollExtent,
          pixels: pixels,
          viewportDimension: viewportDimension,
          axisDirection: axisDirection,
        );

  ///务必重写此方法, 传值过去
  @override
  ScheduleMetrics copyWith({
    double minScrollExtent,
    double maxScrollExtent,
    double pixels,
    double viewportDimension,
    AxisDirection axisDirection,
    double viewportFraction,
  }) {
    return ScheduleMetrics(
      minScrollExtent: minScrollExtent ?? this.minScrollExtent,
      maxScrollExtent: maxScrollExtent ?? this.maxScrollExtent,
      pixels: pixels ?? this.pixels,
      viewportDimension: viewportDimension ?? this.viewportDimension,
      axisDirection: axisDirection ?? this.axisDirection,
      itemWidth: itemWidth ?? this.itemWidth,
      itemSpace: itemSpace ?? this.itemSpace,
      differentCount: differentCount ?? this.differentCount,
      differentWidth: differentWidth ?? this.differentWidth,
    );
  }

  /// The current page displayed in the [PageView].
  double get page {
    if (pixels < itemWidth / 2 + itemSpace / 2) {
      //第一个的宽度, 左边距 + 自身一半
      return 0;
    } else {
//      return (pixels -
//              0.5 * itemWidth +
      return (pixels +
              differentCount * (differentCount - 1) * differentWidth / 2 -
              (differentCount - 1) * (differentCount - 1) * differentWidth) /
          (itemWidth + itemSpace - (differentCount - 1) * differentWidth);
    }
  }

  ///item 宽度
  final double itemWidth;

  ///每次缩小30dp
  final double differentWidth;

  ///缩小数量
  final int differentCount;

  ///item间距
  final double itemSpace;
}

///
class _SchedulePosition extends ScrollPositionWithSingleContext
    implements ScheduleMetrics {
  ///item 宽度
  final double itemWidth;

  ///每次缩小30dp
  final double differentWidth;

  ///缩小数量
  final int differentCount;

  ///item间距
  double _itemSpace;

  double get itemSpace => _itemSpace;

  set itemSpace(double value) {
    if (_itemSpace == value) return;
    final double oldPage = page;
    _itemSpace = value;
//    if (oldPage != null) forcePixels(getPixelsFromPage(oldPage));
  }

  _SchedulePosition({
    this.itemWidth,
    this.differentWidth,
    this.differentCount,
    double itemSpace,
    ScrollPhysics physics,
    ScrollContext context,
    bool keepPage = true,
    ScrollPosition oldPosition,
  })  : assert(keepPage != null),
        _itemSpace = itemSpace,
        super(
          physics: physics,
          context: context,
          keepScrollOffset: keepPage,
          oldPosition: oldPosition,
        );

  ///务必重写此方法, 否则返回类型不对
  @override
  ScheduleMetrics copyWith({
    double minScrollExtent,
    double maxScrollExtent,
    double pixels,
    double viewportDimension,
    AxisDirection axisDirection,
    double viewportFraction,
  }) {
    return ScheduleMetrics(
      minScrollExtent: minScrollExtent ?? this.minScrollExtent,
      maxScrollExtent: maxScrollExtent ?? this.maxScrollExtent,
      pixels: pixels ?? this.pixels,
      viewportDimension: viewportDimension ?? this.viewportDimension,
      axisDirection: axisDirection ?? this.axisDirection,
      itemWidth: itemWidth ?? this.itemWidth,
      itemSpace: itemSpace ?? this.itemSpace,
      differentCount: differentCount ?? this.differentCount,
      differentWidth: differentWidth ?? this.differentWidth,
    );
  }

  ///根据page获取pixel
  ///根据pixels获取page, 需要将page-1
  ///w: itemWidth
  ///c: differentCount
  ///d: differentWidth
  ///n: number
  ///s: itemSpace
  ///P: pixels
  ///p = n * w + n * s - c * (c - 1) * d / 2 - (n - (c - 1)) * (c - 1) * d
  ///page = (p + c * (c - 1) * d / 2 - (c - 1) * (c - 1) * d) / (w + s - (c - 1) * d)
  double getPageFromPixels(double pixels, double itemWidth, double itemSpace,
      double differentWith, int differentCount) {
    if (pixels < itemWidth / 2 + itemSpace / 2) {
      //第一个的宽度, 左边距 + 自身一半
      return 0;
    } else {
//      return (pixels -
//              0.5 * itemWidth +
      return (pixels +
              differentCount * (differentCount - 1) * differentWidth / 2 -
              (differentCount - 1) * (differentCount - 1) * differentWidth) /
          (itemWidth + itemSpace - (differentCount - 1) * differentWidth);
    }
  }

  ///根据page获取pixel
  ///根据pixels获取page, 需要将page-1
  ///w: itemWidth
  ///c: differentCount
  ///d: differentWidth
  ///n: number
  ///s: itemSpace
  ///p = n * w + n * s - c * (c - 1) * d / 2 - (n - (c - 1)) * (c - 1) * d
  double getPixelsFromPage(double page) {
    if (page.round() == 0) {
      return 0;
    }
    //
    return page * itemWidth +
        page * itemSpace -
        differentCount * (differentCount - 1) * differentWidth / 2 -
        (page - (differentCount - 1)) * (differentCount - 1) * differentWidth;
  }

  @override
  double get page {
    assert(
      pixels == null || (minScrollExtent != null && maxScrollExtent != null),
      'Page value is only available after content dimensions are established.',
    );
    return pixels == null
        ? null
        : getPageFromPixels(
            pixels, itemWidth, itemSpace, differentWidth, differentCount);
  }

  @override
  String toString() {
    return '_SchedulePosition{itemWidth: $itemWidth, differentWidth: $differentWidth, differentCount: $differentCount, _itemSpace: $_itemSpace}';
  }
}

///仿写PageView 参照[PageScrollPhysics]
class ScheduleScrollPhysics extends ScrollPhysics {
  /// Creates physics for a [PageView].
  const ScheduleScrollPhysics({ScrollPhysics parent}) : super(parent: parent);

  @override
  ScheduleScrollPhysics applyTo(ScrollPhysics ancestor) {
    return ScheduleScrollPhysics(parent: buildParent(ancestor));
  }

  ///获取当前所在页
  double _getPage(ScrollPosition position) {
    if (position is _SchedulePosition) return position.page;
    return position.pixels / position.viewportDimension;
  }

  double _getPixels(ScrollPosition position, double page) {
    if (position is _SchedulePosition) return position.getPixelsFromPage(page);
    return page * position.viewportDimension;
  }

  ///获取目标像素
  double _getTargetPixels(
      ScrollPosition position, Tolerance tolerance, double velocity) {
    double page = _getPage(position);
    if (velocity < -100)
      page -= 0.0;
    else if (velocity > 100) page += 0.0;
    return _getPixels(position, page.roundToDouble());
  }

  @override
  Simulation createBallisticSimulation(
      ScrollMetrics position, double velocity) {
    print(velocity);
    // If we're out of range and not headed back in range, defer to the parent
    // ballistics, which should put us back in range at a page boundary.
    if ((velocity <= 0.0 && position.pixels <= position.minScrollExtent) ||
        (velocity >= 0.0 && position.pixels >= position.maxScrollExtent))
      return super.createBallisticSimulation(position, velocity);
    final Tolerance tolerance = this.tolerance;
    final double target = _getTargetPixels(position, tolerance, velocity);
    if (target != position.pixels)
      return ScrollSpringSimulation(spring, position.pixels, target, velocity,
          tolerance: tolerance);
    return null;
  }

  @override

  ///是否允许视口隐式更改其滚动位置，以响应对RenderObject.showOnScreen的调用。
  bool get allowImplicitScrolling => false;
}

final ScheduleController _defaultScheduleController = ScheduleController();
const ScheduleScrollPhysics _kSchedulePhysics = ScheduleScrollPhysics();

///思路: 首先设置左边距, 左边距为屏幕一办减去itemWidth/2
///然后计算一屏幕可以放下多少个item, 下一个item的宽度为前一个的宽度
class ScheduleListWidget extends StatefulWidget {
  ///控制器
  final ScheduleController controller;

  ///页面切换事件
  final ValueChanged<int> onPageChanged;

  final ScrollPhysics scrollPhysics;

  ScheduleListWidget({
    Key key,
    this.onPageChanged,
    ScheduleController controller,
    this.scrollPhysics = _kSchedulePhysics,
  })  : controller = controller ?? _defaultScheduleController,
        super(key: key);

  @override
  _ScheduleListWidgetState createState() => _ScheduleListWidgetState();
}

class _ScheduleListWidgetState extends State<ScheduleListWidget> {
  ///当前页面
  int _current = 0;

  @override
  void initState() {
    super.initState();
  }

  @override
  void dispose() {
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return NotificationListener<ScrollNotification>(
      onNotification: (ScrollNotification notification) {
        var metrics2 = notification.metrics;
//        print(
//            "${metrics2.minScrollExtent} - ${metrics2.maxScrollExtent} - ${metrics2.pixels} - ${metrics2.extentAfter}");
        if (notification.depth == 0 &&
            notification is ScrollUpdateNotification) {
          final ScheduleMetrics metrics = notification.metrics;
          final int currentPage = metrics.page.round();
          if (currentPage != _current) {
            setState(() {
              _current = currentPage;
            });
            //如果外部监听了页面切换, 那么通知外部
            if (widget.onPageChanged != null) widget.onPageChanged(currentPage);
          }
        }
        return false;
      },
      child: ListView.builder(
        padding: EdgeInsets.symmetric(horizontal: paddingHorizontal),
        scrollDirection: Axis.horizontal,
        itemCount: 50,
        controller: widget.controller,
        physics: widget.scrollPhysics,
        itemBuilder: (BuildContext context, int index) => AnimatedContainer(
          duration: Duration(milliseconds: 100),
          alignment: Alignment.center,
          constraints: BoxConstraints(
            minWidth: minItemWidth,
          ),
          //计算宽度, item的左右为减小区域
          width: (_current - index).abs() >= widget.controller.differentCount
              ? minItemWidth
              : widget.controller.itemWidth -
                  (_current - index).abs() * widget.controller.differentWidth,
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: <Widget>[
              Image.asset("assets/photo.png"),
              Text(
                "课程列表",
                style: TextStyle(
                    fontSize: (_current - index).abs() >=
                            widget.controller.differentCount
                        ? 10
                        : 16.0 - (_current - index).abs() * 2),
              ),
              Text(
                "拍摄于湖南省",
                style: TextStyle(
                    fontSize: (_current - index).abs() >=
                            widget.controller.differentCount
                        ? 10
                        : 16.0 - (_current - index).abs() * 2),
              ),
            ],
          ),
          margin:
              EdgeInsets.symmetric(horizontal: widget.controller.itemSpace / 2),
        ),
      ),
    );
  }

  ///最小item宽度
  get minItemWidth =>
      widget.controller.itemWidth -
      (widget.controller.differentCount - 1) * widget.controller.differentWidth;

  ///水平边距
  get paddingHorizontal =>
      halfWidth -
      widget.controller.itemWidth / 2 -
      widget.controller.itemSpace / 2;

  ///屏幕半宽
  get halfWidth => MediaQuery.of(context).size.width / 2;

  @override
  void reassemble() {
    super.reassemble();
    setState(() {
      _current = 0;
    });
  }
}
```

### main.dart
```
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
import 'package:flutter_app/schedule_list_widget.dart';

main() => runApp(App());

class App extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: HomePage(),
      theme: ThemeData(primarySwatch: Colors.amber),
    );
  }
}

class HomePage extends StatefulWidget {
  @override
  _HomePageState createState() => _HomePageState();
}

class _HomePageState extends State<HomePage> {
  @override
  void initState() {
    super.initState();
    SystemChrome.setPreferredOrientations(
        [DeviceOrientation.landscapeLeft, DeviceOrientation.landscapeRight]);
  }

  @override
  void dispose() {
    SystemChrome.setPreferredOrientations(DeviceOrientation.values);
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("data"),
      ),
      body: Stack(
        children: <Widget>[
          ScheduleListWidget(),
//          PageView.builder(
//            onPageChanged: (page) {},
//            itemBuilder: (BuildContext context, int index) => AnimatedContainer(
//              duration: Duration(milliseconds: 100),
//              alignment: Alignment.center,
//              constraints: BoxConstraints(
//                minWidth: 160,
//              ),
//              //计算宽度, item的左右为减小区域
//              width: 160,
//              child: Column(
//                mainAxisAlignment: MainAxisAlignment.center,
//                children: <Widget>[
//                  Image.asset("assets/photo.png"),
//                  Text("课程列表"),
//                  Text("拍摄于湖南省"),
//                ],
//              ),
//              margin: EdgeInsets.symmetric(horizontal: 20 / 2),
//            ),
//            itemCount: 20,
//          ),
          Align(
            alignment: Alignment.topCenter,
            child: Container(
              color: Colors.red,
              width: 1,
            ),
          )
        ],
      ),
//      body: Center(
//        child: Text("data"),
//      ),
      floatingActionButton: FloatingActionButton(
        onPressed: () {},
        child: Icon(Icons.ac_unit),
      ),
    );
  }
}
```

**Comment by mofada** [comment]
I'm confused, I need help

**Comment by mofada** [comment]
## flutter doctor -v
```
F:\WorkSpace\Flutter\flutter_app>flutter doctor -v
[√] Flutter (Channel unknown, v1.13.1-pre.1, on Microsoft Windows [Version 10.0.18362.592], locale zh-CN)
    • Flutter version 1.13.1-pre.1 at D:\flutter
    • Framework revision 0f6e91bcca (3 days ago), 2020-02-08 21:31:23 +0800
    • Engine revision 6179380243
    • Dart version 2.7.0 (build 2.7.0-dev.2.1 a4d799c402)


[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at D:\Android\android-sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = D:\Android\android-sdk
    • Java binary at: D:\JetBrains\apps\AndroidStudio\ch-0\192.6165589\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.2.5)
    • Visual Studio at D:\Program Files (x86)\Microsoft Visual Studio\Community
    • Visual Studio Community 2019 version 16.2.29306.81

[√] Android Studio (version 3.5)
    • Android Studio at D:\Android\android-studio
    • Flutter plugin version 42.1.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)

[√] Android Studio (version 3.6)
    • Android Studio at D:\JetBrains\apps\AndroidStudio\ch-0\192.6165589
    • Flutter plugin version 42.2.2
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)

[√] Android Studio (version 3.6)
    • Android Studio at D:\Android\android-studio canary
    • Flutter plugin version 42.2.2
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b04)

[√] Connected device (4 available)
    • Pixel 3 XL • 8BAY0W6W8  • android-arm64  • Android 10 (API 29)
    • Windows    • Windows    • windows-x64    • Microsoft Windows [Version 10.0.18362.592]
    • Chrome     • chrome     • web-javascript • Google Chrome 79.0.3945.130
    • Web Server • web-server • web-javascript • Flutter Tools

• No issues found!
```

**Comment by mofada** [comment]
@TahaTesser  Hello, I stop here now, what am I going to do?

**Comment by TahaTesser** [comment]
Hi @mofada 
I have tested your code multiple times and this issue does exist but there so many things going in your code.
Can you please the isolate ScrollMetrics issue and create a minimal sample code that is just about ScrollMetrics? Also post on Stackoverflow 
Thank you

**Comment by mofada** [comment]
I don't know what to do now, like the value passed to me by ScrollMetrics.pixels is not right, I will try if PageViwe has this problem

**Comment by mofada** [comment]
![image](https://user-images.githubusercontent.com/22286246/74241213-84bfc100-4d16-11ea-8c2c-1065c0cd94bd.png)

If i want to achieve this effect, is there any other way?

**Comment by mofada** [comment]
I seem to have found a problem with PageView, wait a minute, I upload a gif image

**Comment by mofada** [comment]
this is code.

```
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
import 'package:flutter_app/schedule_list_widget.dart';

main() => runApp(App());

class App extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: HomePage(),
      theme: ThemeData(primarySwatch: Colors.amber),
    );
  }
}

class HomePage extends StatefulWidget {
  @override
  _HomePageState createState() => _HomePageState();
}

class _HomePageState extends State<HomePage> {
  int _page;

  @override
  void initState() {
    super.initState();
    SystemChrome.setPreferredOrientations(
        [DeviceOrientation.landscapeLeft, DeviceOrientation.landscapeRight]);
  }

  @override
  void dispose() {
    SystemChrome.setPreferredOrientations(DeviceOrientation.values);
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("data"),
      ),
      body: Stack(
        children: <Widget>[
//          ScheduleListWidget(),
          PageView.builder(
            onPageChanged: (page) => setState(() => _page = page),
            controller: PageController(
                viewportFraction: 160 / MediaQuery.of(context).size.width),
            itemBuilder: (BuildContext context, int index) => AnimatedContainer(
              duration: Duration(milliseconds: 100),
              alignment: Alignment.center,
              constraints: BoxConstraints(
                minWidth: 160,
              ),
              //计算宽度, item的左右为减小区域
              width: 160,
              child: Column(
                mainAxisAlignment: MainAxisAlignment.center,
                children: <Widget>[
                  Image.asset("assets/photo.png"),
                  Text("课程列表"),
                  Text("拍摄于湖南省"),
                ],
              ),
              margin: EdgeInsets.symmetric(horizontal: 20 / 2),
            ),
            itemCount: 50,
          ),
          Align(
            alignment: Alignment.topCenter,
            child: Container(
              color: Colors.red,
              width: 1,
            ),
          ),
          Align(
            child: Text(
              "$_page",
              style: TextStyle(color: Colors.blue, fontSize: 28),
            ),
            alignment: Alignment.topCenter,
          )
        ],
      ),
//      body: Center(
//        child: Text("data"),
//      ),
      floatingActionButton: FloatingActionButton(
        onPressed: () {
          Navigator.of(context).pushAndRemoveUntil(
              MaterialPageRoute(
                  builder: (context) => Container(
                        child: Center(
                          child: Text("data"),
                        ),
                      )),
              (route) => false);
        },
        child: Icon(Icons.ac_unit),
      ),
    );
  }
}

```

**Comment by mofada** [comment]
![image](https://user-images.githubusercontent.com/22286246/74719132-6bff6000-526e-11ea-8e36-7eff70c95446.png)

The maximum distance for each swipe is limited, which is about the width of a screen, so ScrollMetrics.pixels may have this problem in the framework layer, and pixels may not be correct when sliding quickly.
I still have a question, so where to limit its maximum sliding distance each time?

**Comment by mofada** [comment]
![image](https://user-images.githubusercontent.com/22286246/74719799-addcd600-526f-11ea-9694-f381e98f7e94.png)

I'm confused

**Comment by mofada** [comment]
@TahaTesser Can you help me? I still haven't solved it yet.

**Comment by iapicca** [comment]
Hi @mofada 
I'm not really sure about the nature of the problem you are encountering,
but with the standard widget seems to work as intended as in this [sample](https://dartpad.dev/a5ddd92820ceb313be5241c854456e6c)
are you still experiencing the issue with the latest master?
thank you

**Comment by mofada** [comment]
Thank you very much for the case, this is closer to the effect I want, the effect I need is to gradually become smaller from the middle, like this.

![班级-选择课程](https://user-images.githubusercontent.com/22286246/88000416-24884300-cb30-11ea-92a7-11669a86316a.png)

I tried to rewrite according to the code of PageView, the effect is complete, but there will be problems when sliding.If the sliding speed is slow, it is normal, but when my sliding speed is relatively fast, he will have problems.

**Comment by mofada** [comment]
[like this](https://dartpad.dev/79f095aec27a0dda4392c2ed902cc529)

Swipe it slowly is good, when I swipe it quickly there is a problem

**Comment by markusaksli-nc** [comment]
@mofada If this is a problem with with ScrollMetrics or ScrollPosition we would need a much more minimal code sample to demonstrate the issue. It is unclear if the scrolling ballistic offset is due to your personal code or not. 500 lines is a bit much to go through. A simple print to demonstrate the unexpected value when comparing slow scrolling to fast scrolling would be enough.

Could you please reduce your sample to a [minimal complete reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example)?

**Comment by iapicca** [comment]
@mofada 
I believe the problem is not with flutter, but with your implementation of [ScrollPhysics](https://api.flutter.dev/flutter/widgets/PageScrollPhysics-class.html)
on your own [code sample](https://dartpad.dev/79f095aec27a0dda4392c2ed902cc529)
just change a single line as shown below in your  `_ScheduleListWidgetState` class
and everything works as intended

```diff
child: ListView.builder(
        padding: EdgeInsets.symmetric(horizontal: paddingHorizontal),
        scrollDirection: Axis.horizontal,
        itemCount: widget.itemCount,
        controller: widget.controller,
-        physics: widget.scrollPhysics,
+        physics: PageScrollPhysics(),
        itemBuilder: (BuildContext context, int index) => AnimatedContainer(
          duration: Duration(milliseconds: 100),
          alignment: Alignment.center,
          constraints: BoxConstraints(
            minWidth: minItemWidth,
          ),
          //计算宽度, item的左右为减小区域
          width: (_current - index).abs() >= widget.controller.differentCount
              ? minItemWidth
              : widget.controller.itemWidth -
                  (_current - index).abs() * widget.controller.differentWidth,
          child: widget.onScheduleItemBuilder(
            context,
            index,
            _current,
            (_current - index).abs() < widget.controller.differentCount,
          ),
          margin:
              EdgeInsets.symmetric(horizontal: widget.controller.itemSpace / 2),
        ),
      ),
    );
```

**Comment by TahaTesser** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

**Comment by mofada** [comment]
I’m very sorry, I’m currently working on other projects, and I don’t have time to read this question at the moment, thank you very much for your reply

----------

该邮件从移动设备发送







--------------原始邮件--------------
发件人："Taha Tesser "<notifications@github.com&gt;;
发送时间：2020年10月16日(星期五) 晚上9:11
收件人："flutter/flutter" <flutter@noreply.github.com&gt;;
抄送："fada mo "<fada@mofada.cn&gt;;"Mention "<mention@noreply.github.com&gt;;
主题：Re: [flutter/flutter] ScrollMetrics.pixels have problem (#50560)
-----------------------------------

 

 
Closed #50560.
